public/js/src/module/file-manager.js

```diff
@@ -128,6 +154,10 @@ function getCurrentFiles() {
             // TODO: in the future, when browser support increase we can invoke
             // the File constructor to do this.
             newFilename = getFilename( file, this.dataset.filenamePostfix );
+            // If file is resized, get Blob representation of file URL
+            if ( this.dataset.resized && this.dataset.resizedFileUrl && resizedFileBlob ) {
+                file = resizedFileBlob;
+            }
```

Comment:
I see the solution in enketo-core(a5302cd124ec552d52708689e2f9f4c6d5a91f20) and I tried to implement it but the part of the solution in `getCurrentFiles` it's not works as expected. The submitting process won't wait until the promise from `urlToBlob` resolved, as a result, the array files sent will be empty. That's why I tried using the same logic but from different approach by storing the blobs of resized images from `getFileUrl` instead of getting it from `urlToBlob` method when looping files.

Reply:
So maybe we should store resizedFileUrl as a dataURI instead of an objectURL? (There is a method in enketo-express' utils.js that we could move to enketo-core). A dataURI can be synchronously converted to a Blob.

Alternatively, we could perhaps store the resized file in a canvas in the widget, and then we could use the same logic as for drawings in getCurrentFiles. That would be the most elegant I think, but I'm not sure if it will work if the canvas is hidden. 

What do you think?

Reply:
No worries.
I tried using the first option and create a [PR](https://github.com/enketo/enketo-core/pull/651) for that. is that acceptable?

Reply:
Yes, thank you!